server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -419,6 +463,23 @@ private void executeRequest(
             relativeStartNanos,
             System::nanoTime
         );
+        // 2nd executeRequest(), called by 1st executeRequest()
+        final List<SearchRequestOperationsListener> searchListenersList = createSearchListenerList();
+
+        if (originalSearchRequest.isPhaseTookQueryParamEnabled() == SearchRequest.ParamValue.TRUE
+            || (originalSearchRequest.isPhaseTookQueryParamEnabled() == SearchRequest.ParamValue.UNSET
+                && clusterService.getClusterSettings().get(TransportSearchAction.SEARCH_PHASE_TOOK_ENABLED))) {
+            timeProvider.setPhaseTookEnabled(true);
+            searchListenersList.add(timeProvider);
+        }
```

Comment:
You can move this to createSearchListenerList() instead.

Reply:
good idea